content/posts/07-meet-mkmr.md

```diff
@@ -0,0 +1,248 @@
+remote names. For tools that only interact with upstream one can use `--remote=REMOTE`.
+
+The values can be saved in configuration by using `--save` (save and continue operation
+of the requested tool) and `--write` (save and exit), this configuration will exist in
+`${XDG_CONFIG_HOME:-$HOME/.config}/mkmr/config` and takes the name of the current
+directory
+
+![Example Configuration](07-meet-mkmr-05.png)
+
+The configuration takes the form of INI files, the section refers to the repo, and stores
+configuration for the repository generally, such as `--origin`, `--upstream` and `--remote`.
+
+Each tool also has its own per-repository configuration, `[aports mkmr]` as shown above
+denotes configuration for the `mkmr` binary for the `aports` repository.
+
+### mkmr
+
+`mkmr` is the first binary that was written and was the main reason behind starting the
+project itself, email users are used to using git-send-email to send an email to a Mailing
+List, while GitLab expect users to create a fork of the repository you want to contribute to,
+then push a branch to it and create a merge request asking for the branch in your fork to be
+merged into the branch in upstream.
+
+`mkmr` tries to bridge the gap by allowing the user to create the merge request from the
+commandline, by automating pushing a new branch to the user's fork and then creating the
+merge request via the GitLab API.
+
+`mkmr` will prompt before making the merge request, letting you check if your values are
+as expected, this can be skipped by passing `--yes` (and made permanent by passing
+`--save` with it).
+
+![Prompting the user](07-meet-mkmr-02.png)
+
+If one wants to pass a different title, description or labels when creating the merge
+request, one can pass `--edit` (can be made permanent with `--save`), which will open
+`$EDITOR` with instructions, users of `git rebase -i` will notice it is inspired by it.
+
+![Editing the attributes](07-meet-mkmr-03.png)
+
+After checking everything and passing yes to the prompt, the merge request will be created
+
+![Merge Request created](07-meet-mkmr-04.png)
+
+After creating the merge request, the iid of the merge request will be saved in the cache,
+in a file named after the branch of the merge request. This is used by other tools to connect
+a branch name to an specific merge request.
+
+```sh
+$ tree $XDG_CACHE_HOME/mkmr/gitlab.alpinelinux.org/alpine/aports/branches
+/home/enty/var/cache/mkmr/gitlab.alpinelinux.org/alpine/aports/branches
+├── 3.11-busybox
+├── 3.12-alpine-baselayout
+├── 3.12-busybox
+├── arc-theme
+├── busybox
+├── fuse3
+├── perl-module-scandeps
+├── poppler
+├── py3-pytools
+├── py3-validators
+├── screenfetch
+├── strace
+├── tumbler
+├── vips
+└── volumeicon
+
+0 directories, 15 files
+```
+
+### mgmr
+
+The second most important of the `mkmr` tools, `mgmr` handles merging merge requests by
+rebasing them until they are matching upstream, then merging them.
```

Comment:
Maybe it'd be worth mentioning that this only works for FF merges (i.e. without a merge commit)